seqr/migrations/0064_alter_phenotypeprioritization.py

```diff
@@ -0,0 +1,58 @@
+# Generated by Django 3.2.25 on 2024-05-02 17:45
+from django.conf import settings
+from django.db import migrations, models
+import django.utils.timezone
+
+from seqr.models import _slugify
+
+MAX_GUID_SIZE = 30
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('seqr', '0063_dynamicanalysisgroup'),
+    ]
+
+    def update_guids(apps, schema_editor):
+        PhenotypePrioritization = apps.get_model('seqr', 'PhenotypePrioritization')
+        db_alias = schema_editor.connection.alias
+
+        pps = PhenotypePrioritization.objects.using(db_alias).all()
+        for pp in pps:
+            ids_as_str = "%s:%s:%s" % (pp.individual.individual_id, pp.gene_id, pp.disease_id)
+            pp.guid = 'PP%07d_%s' % (pp.id, _slugify(str(ids_as_str)))[:MAX_GUID_SIZE]
```

Comment:
so the reason why we can't use the regular `_compute_guid` functionality for bulk create most of the time is because the `id` does not exist until after the model is created. However here the model and the id already exists so you should be able to just do `pp.guid = pp._compute_guid()` 

Also, when changing around migrations if you want to test them even though you alrwady applied them, its pretty straightforward to unapply and then reapply them in django. What you do is run 
```
./manage.py migrate seqr 0063
```
That will tell django to migrate to whatever number migration you specify, which in this case is the previous migration so it will unapply this migration. Then you can change the migration and then run the migrate command as normal so you can confirm it works

Reply:
🙏 thank you for the tip on unapplying migrations, I would've probably just re-loaded from my most recent db backup

Reply:
It's probably worth adding a reverse_code function then!

Reply:
I'm getting `AttributeError: 'PhenotypePrioritization' object has no attribute '_compute_guid'` when applying, `_compute_guid` must not exist yet on the PhenotypePrioritization model that's in the django app registry. 

I might try doing the data migration in a separate migration that runs after 0064

Reply:
For the reverse migration, you can just do `reverse_code=migrations.RunPython.noop` as theres no need to actually change the guid in reverse because it will be dropped. 

And oh well, I guess we have to leave this as is.  The code you have here for computing the guid will work fine, I don;t think its worth the engineering time to figure out how to get `_compute_guid` to work here

Reply:
agree with both of these